### PR TITLE
Integration test for [VEN-884] Liquidation and [VEN-885] Heal Account scenarios

### PR DESCRIPTION
**Pull request by UrAvgDeveloper**
It includes :

**Liquidation scenario on straight path:** (VEN-884)
- via Comptroller
- via vToken

**Heal Account Scenario:** (VEN-885)
- Revert scenarios
- Success Scenarios

- [x] I have updated the documentation to account for the changes in the code.
- [x] If I added new functionality, I added tests covering it.
- [x] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [x] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: tests/integration/index.ts [hunk 1]
```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
...
+        .withArgs(acc2, bswBorrowAmount, bswBorrowAmount, bswBorrowAmount);
+
+      vBNXPrice = Number(await PriceOracle.getUnderlyingPrice(vBNX.address)) / 10 ** 18;
+      vBSWPrice = Number(await PriceOracle.getUnderlyingPrice(vBSW.address)) / 10 ** 18;
+
+      //Approve more assets for liquidation
+      await BSW.connect(acc1Signer).faucet(bswBorrowAmount);
+      await BSW.connect(acc1Signer).approve(vBSW.address, bswBorrowAmount);
+    });
+    it("Should revert when repay amount on liquidation is not equal to borrow amount", async function () {
```
Comment (coreyar) [inline tests/integration/index.ts:305]: nit: lets add some space between test blocks to make it more readable

File: tests/integration/index.ts [hunk 2]
```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
...
+      const seizeTokens = (bswBorrowAmount * vBSWPrice) / vBNXPrice;
+      await expect(result)
+        .to.emit(vBSW, "LiquidateBorrow")
+        .withArgs(acc1, acc2, bswBorrowAmount, vBNX.address, seizeTokens.toFixed(0));
+
+      const liquidatorBalance = await vBNX.connect(acc1Signer).balanceOf(acc1);
+      expect(liquidatorBalance).to.equal(seizeTokens.toFixed(0));
+    });
+  });
+  describe("Liquidation of user via Vtoken", () => {
```
Comment (coreyar) [inline tests/integration/index.ts:334]: space here

File: tests/integration/index.ts [hunk 3]
```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
...
+      //It should revert when try to borrow more than liquidity
+      await expect(vBSW.connect(acc2Signer).borrow(8e10)).to.be.revertedWithCustomError(
+        Comptroller,
+        "InsufficientLiquidity",
+      );
+      await expect(vBSW.connect(acc2Signer).borrow(bswBorrowAmount))
+        .to.emit(vBSW, "Borrow")
+        .withArgs(acc2, bswBorrowAmount, bswBorrowAmount, bswBorrowAmount);
+
+      vBNXPrice = Number(await PriceOracle.getUnderlyingPrice(vBNX.address)) / 10 ** 18;
```
Comment (coreyar) [inline tests/integration/index.ts:298]: For safety and consistency should we use BigNumber here?

Is there a difference between the two before hooks? They could potentially use a fixture

File: tests/integration/index.ts [hunk 4]
```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
...
+      //borrow
+      //Supply WBTC to market from 2nd account
+      await expect(vBSW.connect(acc1Signer).mint(mintAmount))
+        .to.emit(vBSW, "Mint")
+        .withArgs(acc1, mintAmount, mintAmount);
+      await expect(vBSW.connect(acc2Signer).borrow(bswBorrowAmount))
+        .to.emit(vBSW, "Borrow")
+        .withArgs(acc2, bswBorrowAmount, bswBorrowAmount, bswBorrowAmount);
+
+      vBNXPrice = Number(await PriceOracle.getUnderlyingPrice(vBNX.address)) / 10 ** 18;
```
Comment (coreyar) [inline tests/integration/index.ts:363]: We could use BigNumber and a scaling function here for consistency

File: tests/integration/index.ts [hunk 5]
```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
...
+      const dummyPriceOracle = await smock.fake<PriceOracle>("PriceOracle");
+      dummyPriceOracle.getUnderlyingPrice.whenCalledWith(vBSW.address).returns(convertToUnit("100", 40));
+      dummyPriceOracle.getUnderlyingPrice.whenCalledWith(vBNX.address).returns(convertToUnit("100", 18));
+      await Comptroller.setPriceOracle(dummyPriceOracle.address);
+      //Liquidation
+      await expect(vBSW.connect(acc1Signer).liquidateBorrow(acc2, 201, vBNX.address)).to.be.revertedWith(
+        "LIQUIDATE_SEIZE_TOO_MUCH",
+      );
+    });
+    it("Should revert when liquidation is called through vToken and trying to pay tooMuch", async function () {
```
Comment (coreyar) [inline tests/integration/index.ts:410]:
```suggestion
    it("Should revert when liquidation is called through vToken and trying to pay too much", async function () {
```

File: tests/integration/index.ts [hunk 6]
```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
...
+        .to.emit(vBSW, "Mint")
+        .withArgs(acc1, mintAmount, mintAmount);
+      await expect(vBSW.connect(acc2Signer).borrow(bswBorrowAmount))
+        .to.emit(vBSW, "Borrow")
+        .withArgs(acc2, bswBorrowAmount, bswBorrowAmount, bswBorrowAmount);
+
+      vBNXPrice = Number(await PriceOracle.getUnderlyingPrice(vBNX.address)) / 10 ** 18;
+      vBSWPrice = Number(await PriceOracle.getUnderlyingPrice(vBSW.address)) / 10 ** 18;
+
+      //Approve more assets for liquidation
```
[the remark this answers is not in the document]
Comment (UrAvgDeveloper) [reply tests/integration/index.ts:366]: I checked prettier doc, but could not able to find options to lint comments, for now i did it manually in code, If you have any knowledge about it, please do let me know.

File: tests/integration/index.ts (hunk 1 above)
Comment (UrAvgDeveloper) [reply tests/integration/index.ts:316]: Fixed

File: tests/integration/index.ts (hunk 2 above)
Comment (UrAvgDeveloper) [reply tests/integration/index.ts:347]: Fixed

File: tests/integration/index.ts (hunk 3 above)
Comment (UrAvgDeveloper) [reply tests/integration/index.ts]: BigNumber is fixed
There are multiple tickets scenarios in one block and have slightly changes in each before.

File: tests/integration/index.ts (hunk 4 above)
Comment (UrAvgDeveloper) [reply tests/integration/index.ts]: BigNumber is fixed.
What does scaling function explicitly mean?
Removed all redundant calculation of price in code.

File: tests/integration/index.ts [hunk 7]
```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
...
+      await BSW.connect(acc1Signer).faucet(convertToUnit("1", 18));
+      await BSW.connect(acc1Signer).approve(vBSW.address, convertToUnit("1", 18));
+    });
+    it("Should revert when liquidation is called through vToken and does not met minCollateral Criteria", async function () {
+      await expect(vBSW.connect(acc1Signer).liquidateBorrow(acc2, bswBorrowAmount, vBSW.address))
+        .to.be.revertedWithCustomError(Comptroller, "MinimalCollateralViolated")
+        .withArgs("100000000000000000000", "15999000000");
+    });
+    it("Should revert when liquidation is called through vToken and no shortfall", async function () {
+      //Mint and Incrrease collateral of the user
```
[the remark this answers is not in the document]
Comment (UrAvgDeveloper) [reply tests/integration/index.ts]: Fixed

File: tests/integration/index.ts (hunk 5 above)
Comment (UrAvgDeveloper) [reply tests/integration/index.ts]: Fixed

File: tests/integration/index.ts (hunk 4 above)
Comment (coreyar) [reply tests/integration/index.ts]: We could write a helper function similar to `convertToUnit` that scales a number down to its decimal based on a provided exponent.

**Pull request merged**